aicspylibczi/CziFile.py

```diff
@@ -261,15 +264,27 @@ def read_subblock_metadata(self, **kwargs):
 
         Returns
         -------
-        [str]
-            an array of stings containing the specified subblock metadata
-
+        [(dict, str)] if unified_xml is False
+            an array of tuples containing a dimension dictionary and the corresponding subblock metadata
+        lxml.etree.Element if unified_xml is True
+            an lxml document containing the requested subblock metadata.
         """
         plane_constraints = self.czilib.DimCoord()
         [plane_constraints.set_dim(k, v) for (k, v) in kwargs.items() if k in CziFile.ZISRAW_DIMS]
         m_index = self._get_m_index_from_kwargs(kwargs)
-
-        return self.reader.read_meta_from_subblock(plane_constraints, m_index)
+        subblock_meta = self.reader.read_meta_from_subblock(plane_constraints, m_index)
+        if not unified_xml:
+            return subblock_meta
+        root = etree.Element("Subblocks")
+        for pair in subblock_meta:
+            new_element = etree.Element("Subblock")
+            for dim, number in pair[0].items():
+                new_element.set(dim, str(number))
+            if 'S' not in pair[0]:
+                new_element.set('S', "0")
```

Comment:
is this still modifying the xml?

Reply:
Yup, this is intentional. Essentially the subblock metadata chunks are missing context, as a consequence, there is no "native" way to export the context (the dimensions of the subblock). My solution is 
'''
<Subblocks>
  <Subblock B="0" C="2" S="2" Z="3">
    <METADATA>
            ...
    </METADATA>
  </Subblock>'''

The METADATA tag and everything inside is untouched subblock metadata. It was the most minimal way to inject the context. I intend to post something on ImageSC asking Zeiss how they recommend maintaining context for the subblock metadata.